pydatastructs/graphs/algorithms.py

```diff
@@ -548,3 +548,36 @@ def strongly_connected_components(graph, algorithm):
         "isn't implemented for finding strongly connected components."
         %(algorithm, graph._impl))
     return getattr(algorithms, func)(graph)
+
+
+def random_walk(G,path_length,rand=random.Random()start=None):
+    '''
+    G is adjacency list 
```

Comment:
What if adjacency matrix representation or implicit graph representation is used? Do we need different implementer functions for some representations of graphs?